### ISSUE TITLE
removeUpcomingTracks is removing the currently playing item when an item is already in the queue on Android

### ISSUE THREAD

**Issue by mitchdowney** [doublesymmetry/react-native-track-player#1953]
**Describe the Bug**
removeUpcomingTracks is removing the currently playing item when an item is already in the queue on Android.

**Steps To Reproduce**
Steps to reproduce on Android:
1) Add multiple tracks to the queue and press play.
2) Call removeUpcomingTracks.
3) The currently playing item is removed unexpectedly, and the next item begins playing.

**Code To Reproduce**
I need to work on this...

**Replicable on Example App?**
I need to work on this...

**Environment Info:**
```
System:
    OS: macOS 13.2.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz
    Memory: 1.31 GB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.1 - ~/.nvm/versions/node/v16.17.1/bin/node
    Yarn: 1.22.18 - ~/.yarn/bin/yarn
    npm: 6.14.17 - ~/Repos/podverse-rn/node_modules/.bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8512546
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.9 - /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.66.5 => 0.66.5 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

react-native-track-player
3.2.0-bb749e5d286737681ba8907dfd287480eb1cec45

Real device (haven't tested simulator)

Android 13

**How I can Help**
What can you do to help resolve this?
Try to make a reproduction demo.

Have you investigated the underlying JS or Swift/Android code causing this bug?
Not really, sorry.

**Comment by mitchdowney** [comment]
I wonder if this has something to do with the reverse sort happening in removeUpcomingItems in KotlinAudio? It is a recent code change, and if we're removing items in reverse order, then perhaps it is accidentally removing the first item in the queue.

https://github.com/doublesymmetry/KotlinAudio/pull/50/files#diff-f61629c9e3e84e4bd20e36b561ae95c1530db3aa5c76665c9f4667a4edceae9fR166

**Comment by mitchdowney** [comment]
Bug demo branch here: https://github.com/podverse/react-native-track-player/tree/1953-removeUpcomingTracks-android-bug

1. Restart app on Android.
2. Press play.
3. Press removeUpcomingTracks.
**Error:** the currently playing track will be skipped.

![image](https://user-images.githubusercontent.com/2262919/225519775-a2da96be-0b2c-43ed-a100-b69676f073b3.png)

**Comment by puckey** [comment]
Should be fixed by https://github.com/doublesymmetry/KotlinAudio/pull/67

**Comment by jspizziri** [comment]
https://github.com/doublesymmetry/KotlinAudio/pull/67 has been merged, should be available on jitpack soon and we can bump the version in RNTP.

**Comment by mitchdowney** [comment]
@jspizziri re: your message in Discord to try submiting a PR to RNTP bumping the KotlinAudio version, I took a stab at it, but I think I did it wrong...

https://github.com/doublesymmetry/react-native-track-player/pull/1968

1) I saw the commit was available in jitpack, so I specified the commit hash for the kotlinaudio dependency in build.grade. According to my tests locally, Puckey's fix was applied and worked when I did this. Is commit hash ok for this RC?

2) I updated the changelog, but wasn't sure how to reference the KotlinAudio PR properly.

3) Is ok if I mix the KotlinAudio upgrade and RC release in the same PR? Seems like for cleanliness I should have made 2. I can do that if you'd like.
